tests/run-make/branch-protection-check-IBT/Makefile

```diff
@@ -1,21 +0,0 @@
-# Check for GNU Property Note
-
-include ../tools.mk
-
-# How to run this
-# python3 x.py test --target x86_64-unknown-linux-gnu  tests/run-make/branch-protection-check-IBT/
-
-# only-x86_64
-
-# ignore-test
-# FIXME(jieyouxu): This test never runs because the `ifeq` check on line 17
-# compares `x86` to `x86_64`, which always evaluates to false.
-# When the test does run, the compilation does not include `.note.gnu.property`.
-# See https://github.com/rust-lang/rust/pull/126720 for more information.
-
-all:
-ifeq ($(filter x86,$(LLVM_COMPONENTS)),x86_64)
-	$(RUSTC) --target x86_64-unknown-linux-gnu -Z cf-protection=branch -L$(TMPDIR) -C link-args='-nostartfiles'  -C save-temps  ./main.rs -o $(TMPDIR)/rsmain
```

Comment:
Discussion: I used `cross` to try to reproduce this, and lo-and-behold, there is no `.note.gnu.property`:

```
PS E:\Repos\test\huh> cat .\target\x86_64-unknown-linux-gnu\debug\.fingerprint\huh-1e1959310e6cde68\bin-huh.json
{"rustc":11041252367277408880,"features":"[]","declared_features":"[]","target":3507701825157409998,"profile":6929766844333714335,"path":10602529704205407992,"deps":[],"local":[{"CheckDepInfo":{"dep_info":"x86_64-unknown-linux-gnu/debug/.fingerprint/huh-1e1959310e6cde68/dep-bin-huh"}}],"rustflags":["-Z","cf-protection=branch","-C","link-args=-nostartfiles"],"metadata":7797948686568424061,"config":2202906307356721367,"compile_kind":9018714775688763800}
PS E:\Repos\test\huh> readelf -n .\target\x86_64-unknown-linux-gnu\debug\huh

Displaying notes found in: .note.gnu.build-id
  Owner                Data size        Description
  GNU                  0x00000014       NT_GNU_BUILD_ID (unique build ID bitstring)
    Build ID: fa9d90e0ef28addf979dfd4645b35a4a06308327
PS E:\Repos\test\huh>
```

there is `.note.gnu.build-id`. I'm suspecting maybe the Intel CET landscape changed to no longer require this note...?

Reply:
Rejyr got to display the gnu property note by using the following commands:

```
$ rustc +stable main.rs -o main
$ llvm-readobj --elf-output-style=GNU -nW main
```

Could this have something to do with `--elf-output-style=GNU`? Or perhaps with the extra flags being passed to rustc?

Reply:
Ah good point. I need to double-check this with the people who knows about Intel IBT (not to be confused with Arm's BTI). Thanks for the info.not 

---

Remark (for myself):

- Initial issue: #103001
- Initial PR: #110304

Relevant docs:

### Arm
BTI: Branch Target Identification
Docs: <https://developer.arm.com/documentation/102433/latest/Applying-these-techniques-to-real-code>

### Intel x86_64
IBT: Indirect Branch Tracking
Docs: <https://edc.intel.com/content/www/us/en/design/ipla/software-development-platforms/client/platforms/alder-lake-desktop/12th-generation-intel-core-processors-datasheet-volume-1-of-2/006/indirect-branch-tracking/>

Reply:
> Could this have something to do with `--elf-output-style=GNU`?

That seems a bit weird, because in my case I was compiling on a x86_64 host for a x86_64 target, so surely `--elf-output-style` can't be `arm`... right?

Reply:
![Screenshot 2024-08-29 194500](https://github.com/user-attachments/assets/bb919bd5-93cb-403f-9a4a-e761610c376d)

no repro in a docker container w/ `x86_64-gnu` either.